components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java

```diff
@@ -6424,4 +6425,36 @@ private void audit(String action, String data, String result) {
 
         AUDIT_LOG.info(String.format(AUDIT_MESSAGE, loggedInUser, action, data, result));
     }
+
+    public List<TrustedApp> getTrustedApps(String platformType) throws IdentityApplicationManagementException {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Getting trusted app details for platform type: " + platformType);
+        }
+
+        List<TrustedApp> trustedApps = new ArrayList<>();
+
+        try (Connection connection = IdentityDatabaseUtil.getDBConnection(false);
+             PreparedStatement statement = connection.prepareStatement(
+                     ApplicationMgtDBQueries.LOAD_TRUSTED_APPS_BY_PLATFORM_TYPE)) {
+            statement.setString(1, platformType);
+            statement.execute();
+
+            try (ResultSet resultSet = statement.executeQuery()) {
+                while (resultSet.next()) {
+                    TrustedApp app = new TrustedApp();
+                    app.setPlatformType(platformType);
+                    app.setAppIdentifier(resultSet.getString(1));
+                    app.setThumbprints(resultSet.getString(2));
+                    app.setIsFIDOTrusted(resultSet.getBoolean(3));
+                    app.setIsTWAEnabled(resultSet.getBoolean(4));
+                    trustedApps.add(app);
+                }
+            }
+        } catch (SQLException e) {
+            String msg = "Error occurred while retrieving trusted app list for %s platform.";
+            throw new IdentityApplicationManagementException(String.format(msg, platformType), e);
+        }
+        return trustedApps;
+    }
```

Comment:
Creating app entities for separate SPs. Should we only separate out based on tenant?